tests/certification/state/azure/tablestorage/tablestorage_test.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tablestorage_test
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+
+	state "github.com/dapr/components-contrib/state"
+	table "github.com/dapr/components-contrib/state/azure/tablestorage"
+	"github.com/dapr/components-contrib/tests/certification/embedded"
+	"github.com/dapr/components-contrib/tests/certification/flow"
+	"github.com/dapr/go-sdk/client"
+
+	"github.com/dapr/components-contrib/secretstores"
+	secretstore_env "github.com/dapr/components-contrib/secretstores/local/env"
+	secretstores_loader "github.com/dapr/dapr/pkg/components/secretstores"
+
+	"github.com/dapr/components-contrib/tests/certification/flow/sidecar"
+	state_loader "github.com/dapr/dapr/pkg/components/state"
+	"github.com/dapr/dapr/pkg/runtime"
+	dapr_testing "github.com/dapr/dapr/pkg/testing"
+	"github.com/dapr/kit/logger"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	sidecarNamePrefix       = "table-sidecar-"
+	certificationTestPrefix = "stable-certification-"
+)
+
+func TestAzureTableStorage(t *testing.T) {
+	log := logger.NewLogger("dapr.components")
+	ports, err := dapr_testing.GetFreePorts(2)
+	assert.NoError(t, err)
+
+	currentGrpcPort := ports[0]
+	currentHTTPPort := ports[1]
+
+	basicTest := func(ctx flow.Context) error {
+		client, err := client.NewClientWithPort(fmt.Sprint(currentGrpcPort))
+		if err != nil {
+			panic(err)
+		}
+		defer client.Close()
+
+		// save state, default options: strong, last-write
+		err = client.SaveState(ctx, "statestore-basic", certificationTestPrefix+"key1", []byte("certificationdata"), nil)
+		assert.NoError(t, err)
+
+		// get state
+		item, err := client.GetState(ctx, "statestore-basic", certificationTestPrefix+"key1", nil)
+		assert.NoError(t, err)
+		assert.Equal(t, "certificationdata", string(item.Value))
+
+		// delete state
+		err = client.DeleteState(ctx, "statestore-basic", certificationTestPrefix+"key1", nil)
+		assert.NoError(t, err)
+
+		return nil
+	}
+
+	NonExistingTableTest := func(ctx flow.Context) error {
+		client, err := client.NewClientWithPort(fmt.Sprint(currentGrpcPort))
```

Comment:
Don't duplicate this function. Just simply use the existing function and add a second step for the deletion of the table only. In your original function you can parameterize that function so it takes as input the name of the state store to use!